cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -199,3 +236,45 @@ def test_flavor_details(appliance, soft_assert, private_flavor):
 
     soft_assert(view.entities.relationships.get_text_of('Cloud Provider') ==
                 private_flavor.provider.name)
+
+
+@pytest.mark.rfe
+@pytest.mark.ignore_stream('5.9')
+def test_create_instance_with_private_flavor(instance_with_private_flavor, provider, soft_assert):
+    view = navigate_to(instance_with_private_flavor, 'Details')
+    prov_data = provider.data['provisioning']
+    power_state = view.entities.summary('Power Management').get_text_of('Power State')
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = view.entities.summary('Relationships').get_text_of('VM Template')
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    flavors = [f.name for f in provider.mgmt.api.flavors.list()]
+    flavor = view.entities.summary('Relationships').get_text_of('Flavor')
+    soft_assert(flavor in flavors)
```

Comment:
Required:
I don't think here you need to use `soft_assert` from the provider side. and `L252` you can use as flavor has `all` method. More stick to `cfme` than a provider. what do you think?
`flavor in appliance.collections.cloud_flavors.all()`

Reply:
The logic was not to compare cfme to itself but if it is OK to use lavor in appliance.collections.cloud_flavors.all(), I am OK with that